openlibrary/plugins/upstream/account.py

```diff
@@ -414,6 +414,40 @@ def render_error(self, error_key, i):
         f.note = get_login_error(error_key)
         return render.login(f)
 
+    def get_post_login_action(self, web):
+        """Extract the 'action' parameter from the query string.
+        Return the action if it is valid, else return None
+        """
+        qs = web.ctx.env.get('QUERY_STRING', "")
+        parsed_query = parse_qs(qs)
+        action = parsed_query.get("action", [None])[0]
+        if action:
+            return action
+        return action
```

Comment:
I feel like we may not need the function `get_post_login_action` because the action is already available within `i = web.input(action=...)`

https://github.com/internetarchive/openlibrary/pull/10365/files#diff-ef23faa0d3112e7eef1f574a58b0710f9604f911f5a57e795e45c13c4b91fa19R465